course work 2 ready

sergeywhite86 opened the pull request.

sergeywhite86 committed:
init commit

artem-petrosyan commented on src/main/java/com/white/skyprocoursework2/repository/repo/InMemoryDb.java:
```diff
@@ -0,0 +1,40 @@
+public class InMemoryDb {
+
+    public final Set<Question> JAVA_QUESTIONS = new HashSet<>();
+    public final Set<Question> MATH_QUESTIONS = new HashSet<>();
```
Идея неплохая, но можно без spirng'а обойтись. Сделать поля static и заменить postContruct на статический блок инициализации.

Но если так оставлять, то стоит закрыть поля, изменить нейминг на camelCase и добавить геттеры

artem-petrosyan commented on src/main/java/com/white/skyprocoursework2/entity/Question.java:
```diff
@@ -0,0 +1,10 @@
+
+import lombok.*;
+
+@Data
```
На будущее учти, что на классы с аннотацией entity лучше не вешать аннотацию data

artem-petrosyan commented on src/main/java/com/white/skyprocoursework2/repository/impl/JavaQuestionRepository.java:
```diff
@@ -0,0 +1,58 @@
+
+    @Override
+    public Question getRandomQuestion() {
+        List<Question> questionList = new ArrayList<>(findAll());
+        Collections.shuffle(questionList);
+        return questionList.get(0);
```
Не лучший способ получения случайного вопроса.
По сути ты копируешь список, для того, чтобы получить лишь один элемент. 
Это не выгодно по памяти.
Можно через класс Random сгененрировать случайное число от 0 до размера коллекции и взять элемент по этому счёту.

В MathQuestionRepository аналогично.

artem-petrosyan commented on src/main/java/com/white/skyprocoursework2/controller/ExamController.java:
```diff
@@ -0,0 +1,22 @@
+
+    @GetMapping("/get/{amount}")
+    public Collection<Question> getQuestions(@PathVariable int amount){
+       return   examinerService.getQuestions(amount);
```
Лишние пробелы.

sergeywhite86 committed:
1 - поменял структуру хранения БД на List(для возможности доступа по индексу к данным и получения случайного вопроса через объект класса Random)
2 - сделал правильный нейминг в объектах хранилища(InMemoryDb), инкапсулировал поля.
3 - поменял аннотацию @Data в Question
4 - убрал лишние пробелы в ExamController

sergeywhite86 committed:
Спасибо за подсказку со стримами и методом skip();
